File: src/com/test/serializable/People.java
package com.test.serializable;

import java.io.Serializable;

/**
 * Created by chenfeiyue on 16/1/11.
 */
public class People implements Serializable{

    private static final long serialVersionUID = 2L;

    private String name;
    private int age;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

	@Override
	public String toString() {
		return "People [name=" + name + ", age=" + age + "]";
	}
    
    
}

File: src/com/test/kotlin/People.kt
package com.test.kotlin

import java.math.BigDecimal
import kotlin.reflect.jvm.internal.impl.javax.inject.Inject

/**
 * Created by chenfeiyue on 17/7/28.
 * Description:
 */

// 常规用法
class People1(name: String) {
}

// 当主构造函数有注解或者可见性修饰符，需加 constructor 关键字
class People2 public @Inject constructor(name: String) {
}

//若主构造函数中，不进行初始化, 可放在init{}中
class People3(name: String) {
    val name: String

    init {
        println("initialize")
        this.name = name
    }
}

// 如果类有一个主构造函数（无论有无参数），每个次构造函数需要直接或间接委托给主构造函数，用this关键字
class People4 {
//    constructor() {
//        println("constructor")
//    }
    constructor(name: String) {
println(name)
    }
    constructor(name: String, age: Int) {

    }
}


class People5 {
    private var name: String = ""
//        get() : String {
//            return name
//        }
//        set(value) {
//            name = value
//        }
    private var age : Int = 0
//        get() : Int {
//            return age
//        }
//        set(value) {
//            age = value
//        }

}